packages/core/framework/src/http/utils/get-query-config.ts

```diff
@@ -29,27 +28,15 @@ export function prepareListQuery<T extends RequestQueryFields, TEntity>(
   queryConfig: QueryConfig<TEntity> = {}
 ) {
   // TODO: this function will be simplified a lot once we drop support for the old api
-  const { order, fields, limit = 50, expand, offset = 0 } = validated
-  let {
-    allowed = [],
-    defaults = [],
-    defaultFields = [],
-    defaultLimit,
-    allowedFields = [],
-    allowedRelations = [],
-    defaultRelations = [],
-    isList,
-  } = queryConfig
-
-  allowedFields = allowed.length ? allowed : allowedFields
-  defaultFields = defaults.length ? defaults : defaultFields
+  const { order, fields, limit = 50, offset = 0 } = validated
+  let { allowed = [], defaults = [], defaultLimit, isList } = queryConfig
```

Comment:
todo: the defaultLimit seems to never be used because we have a default value on limit, I would suggest to inverse the default value and set it to the default limit instead

Reply:
Done